cluster-autoscaler/core/scale_down.go

```diff
@@ -382,9 +450,11 @@ func (sd *ScaleDown) TryToScaleDown(allNodes []*apiv1.Node, pods []*apiv1.Pod, p
 			errors.CloudProviderError,
 			errCP)
 	}
-	coresTotal, memoryTotal := calculateCoresAndMemoryTotal(nodesWithoutMaster, currentTime)
-	coresLeft := coresTotal - resourceLimiter.GetMin(cloudprovider.ResourceNameCores)
-	memoryLeft := memoryTotal - resourceLimiter.GetMin(cloudprovider.ResourceNameMemory)
+
+	scaleDownResourcesLeft, err := computeScaleDownResourcesLeftLimits(nodesWithoutMaster, resourceLimiter, currentTime)
+	if err != nil {
+		return ScaleDownError, err.AddPrefix("failed to obtain nodes resources: ")
```

Comment:
Actually code new code and old code will behave exactly the same in terms of returning from `TryToScaleDown` in case of an error here.

Note the section above (which is removed in subsequent fixup commit)
```golang
 	resourceLimiter, errCP := sd.context.CloudProvider.GetResourceLimiter()
 	if errCP != nil {
 		return ScaleDownError, errors.ToAutoscalerError(
 			errors.CloudProviderError,
 			errCP)
 	}
```
The `computeScaleDownResourcesLeftLimits` will return an error only in exactly same situation.

But even without it I strongly feel we should not mask errors. I would rather see cluster-autoscaler not performing scale-down on some misbehaving cluser. And add specific patch (in cluster autoscaler or in other place on case-by-case basis), to mitigate that specific situation, rather than blindly ignore errors and proceed.

Reply:
Good point, although I'd rather not exit on error from this function regardless. Can we retrieve resource limiter earlier, and pass it as a parameter?

Mitigation (by limiting impact of errors) is exactly what I mean here. This doesn't need to cause autoscaling downtime, so it probably shouldn't. If observability is a concern, perhaps we can emit an event or publish a metric that will alert us when this happens?